src/nile/common.py

```diff
@@ -93,16 +95,38 @@ def parse_information(x):
     return address, tx_hash
 
 
-def stringify(x):
+def stringify(x, process_short_strings=False):
     """Recursively convert list or tuple elements to strings."""
     if isinstance(x, list) or isinstance(x, tuple):
-        return [stringify(y) for y in x]
+        return [stringify(y, process_short_strings) for y in x]
     else:
+        if process_short_strings and is_string(x):
+            return str(str_to_felt(x))
         return str(x)
 
 
 def prepare_params(params):
     """Sanitize call, invoke, and deploy parameters."""
     if params is None:
         params = []
-    return stringify(params)
+    return stringify(params, True)
+
+
+def is_string(param):
+    """Identify a param as string if is not int or hex."""
+    is_int = True
+    is_hex = True
+
+    # converting to integer
+    try:
+        int(param)
+    except Exception:
+        is_int = False
+
+    # converting to hex
+    try:
+        int(param, 16)
+    except Exception:
+        is_hex = False
```

Comment:
Should we also include the check `param.startswith("0x")`? Otherwise, I think shorts strings like "bad" will be interpreted as hex and not convert to a felt

Reply:
Good catch! Agree.